clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/DefaultCacheContainer.java

```diff
@@ -91,13 +91,15 @@ public void stop() {
 
     @Override
     public <K, V> Cache<K, V> getCache(String cacheName, boolean createIfAbsent) {
-        return this.wrap(this.cm.<K, V>getCache(cacheName, createIfAbsent));
+        Cache<K, V> cache = this.cm.<K, V>getCache(cacheName, createIfAbsent);
+        return (cache != null) ? this.wrap(cache) : null;
     }
 
     @Deprecated
     @Override
     public <K, V> Cache<K, V> getCache(String cacheName, String configurationTemplate, boolean createIfAbsent) {
-        return this.wrap(this.cm.<K, V>getCache(cacheName, configurationTemplate, createIfAbsent));
+        Cache<K, V> cache = this.cm.<K, V>getCache(cacheName, configurationTemplate, createIfAbsent);
+        return (cache != null) ? this.wrap(cache) : null;
```

Comment:
Wouldn't it be better to check this in the "wrap" method?
With this approach you are assuming implicit behaviour on the delegate manager.

Reply:
getCache(...) can only return null in the createIfAbsent variants.

Reply:
But... aren't you inferring that knowledge by assuming how "cm" works?

Reply:
> But... aren't you inferring that knowledge by assuming how "cm" works?

This behavior is explicitly defined in the EmbeddedCacheManager contract.  As an implementor (via decoration) of that contract, this knowledge is to be expected.